openedx/core/djangoapps/catalog/utils.py

```diff
@@ -135,3 +138,41 @@ def get_programs_with_type(types=None):
             programs_with_type.append(program_with_type)
 
     return programs_with_type
+
+
+def get_catalog_course_runs():
+    """
+    Retrieve all the course runs from the catalog service.
+
+    Returns:
+        list of dict with each record representing a course run.
+    """
+    catalog_integration = CatalogIntegration.current()
+    course_runs = []
+    if catalog_integration.enabled:
+        try:
+            user = User.objects.get(username=catalog_integration.service_username)
+        except User.DoesNotExist:
+            log.error(
+                '[%s] user: %s does not exist.',
```

Comment:
This message will come out looking a little strange. I would replace it with something like:
```
log.exception('Catalog service user with username [%s] does not exist. Course runs will not be retrieved.', catalog_integration.service_username)
```